### ISSUE TITLE
Merge all utility C++ libraries into GPI

### ISSUE THREAD

**Issue by ktbarrett** [cocotb/cocotb#2274]
After #2268 is merged. All the reasons that libgpilog, libembed, and libcocotb_utils were separate objects will be gone and they should be merged into the GPI library, then libcocotb will gain a dependency on libgpi (which was not previously possible).

**Comment by ktbarrett** [comment]
Closed by #5092.
